### PR TITLE
update maintenance guide with recent news

### PR DESCRIPTION
**Pull request by Byron**
This might be interesting to our new maintainer,  @jongiddy - welcome  🎉.

### CODE REVIEW

File: MAINTENANCE.md
Comment (jongiddy) [inline MAINTENANCE.md:20]: I would like an introductory paragraph to describe why there are two release processes, and which one to use by default.
```suggestion
#### Process
We are moving to using `cargo smart-release` for releases, but it still needs some polishing. [Some criteria to choose which one to use.]
##### `cargo publish`
...
##### `cargo smart-release`
...
```

File: MAINTENANCE.md
Comment (jongiddy) [inline MAINTENANCE.md:44]: Does this create a second commit or update the commit from step 3?

Comment (jongiddy) [inline MAINTENANCE.md:44]:
```suggestion
4. `cargo smart-release -e --update-crates-index` - double-check that the release version is sensible, and if not, edit `Cargo.toml` to specify a sensible release.
```

File: MAINTENANCE.md
Comment (jongiddy) [inline MAINTENANCE.md:45]: Is this due to use of `v...` tag? Any way to configure `smart-release` to use tags without the `v`?

File: MAINTENANCE.md
Comment (jongiddy) [inline MAINTENANCE.md:47]: It looks like the release for 1.0.31 was not populated. Can you describe the steps more clearly here, perhaps by doing them for 1.0.31?

**Comment by jongiddy** [comment]
In the `cargo publish` section there is an instruction to `get it merged` which implies getting approval to merge. There should be something similar in the `cargo smart-release` section. In both sections, maybe it should read `get the PR approved and merged`.

File: MAINTENANCE.md
Comment (Byron) [reply MAINTENANCE.md:20]: Actually, this could be up to discussion if having two processes even makes sense.
One day I just 'felt' very strongly about wanting to use `cargo smart-release` but didn't think of the incompatibility regarding the tags, and the other problems that might arise from a PR-based workflow which `cargo smart-release` isn't designed around (so `gh` invocations might fail).

Even though Changelogs are nice, in the end, it might be easier to remove it and call it a failed experiment - I'd actually vote for that.

**Comment by Byron** [comment]
Thanks for the review, @jongiddy , I am very happy to not be left alone here anymore :).

My first suggestion is to remove the remnants of the `cargo smart-release` experiment and go back to doing it like previously described. If you agree, I invite you to make the changes you'd find useful directly and push them to this branch - that's easier and faster as well.

File: MAINTENANCE.md
Comment (jongiddy) [reply MAINTENANCE.md:20]: Reverting the changes from the Process header down would be clearer. For changelogs, is it possible to add `cargo changelog` to the `cargo publish` workflow?

Comment (Byron) [reply MAINTENANCE.md:20]: That’s definitely possible.

**Comment by Byron** [comment]
Now that you are maintainer @jongiddy , you will probably be able to push into this branch to complete it.
I guess it's my way of saying "hi, and welcome" :).

**Comment by jongiddy** [comment]
I removed the smart-release section and revised the text of the other sections.

**Review by Byron — COMMENTED** [review]
LGTM! Please feel free to approve and merge.

**Review by jongiddy — APPROVED** [review]

**Comment by jongiddy** [comment]
@Byron The repo has auto-merge set, so the PR is automatically merged by you when any maintainer approves. That setting may not make as much sense with multiple maintainers.

**Comment by Byron** [comment]
I think that might have been set for this PR only, I wouldn't expect that to happen next time unless it's explicitly enabled on a per PR basis.

**Pull request merged**
